kroxylicious-filters/kroxylicious-sasl-inspection/src/main/java/io/kroxylicious/filters/sasl/inspection/SaslObserver.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filters.sasl.inspection;
+
+import org.apache.kafka.common.errors.AuthenticationException;
```

Comment:
Do we want to couple this interface to this Kafka exception? On the one hand it seems harmless enough. But I worry that incrementally such decisions are going to cause pain in the long run.